service-13/LargeMessageTransaction.asn1

```diff
@@ -0,0 +1,45 @@
+--!
+--! Copyright (C) 2017 N7 Mobile sp. z o. o.
+--! Contact: http://n7mobile.pl/Space
+--!
+--! This file is part of ASN.1/ACN PUS-C Components Library.
+--!
+--! Library was developed under a programme and funded by
+--! European Space Agency.
+--!
+--! This Library is free software: you can redistribute it and/or modify
+--! it under the terms of the GNU General Public License as published by
+--! the Free Software Foundation, either version 3 of the License, or
+--! (at your option) any later version.
+--!
+--! This Library is distributed in the hope that it will be useful,
+--! but WITHOUT ANY WARRANTY; without even the implied warranty of
+--! MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+--! GNU General Public License for more details.
+--!
+--! You should have received a copy of the GNU General Public License
+--! along with this program.  If not, see <http://www.gnu.org/licenses/>.
+--!
+
+
+LargeMessageTransaction DEFINITIONS AUTOMATIC TAGS ::= BEGIN
+EXPORTS ALL;
+IMPORTS
+    PUSC-UINT32 FROM BasicTypes;
+
+    LargeMessageTransactionIdentifierType ::= PUSC-UINT32
+
+    PartReportGeneric { UnsignedInteger, PartType } ::= SEQUENCE
+    {
+        largeMessageTransactionIdentifier LargeMessageTransactionIdentifierType, -- must be unique
+        partSequenceNumber UnsignedInteger,
+        part PartType
+    }
+
+    maxPartSizeTransfer INTEGER ::= 4096
+
+    PartSizeOctedStringType ::= OCTET STRING (SIZE(maxPartSizeTransfer))
+
+    UplinkPartReport ::= PartReportGeneric {PUSC-UINT32, PartSizeOctedStringType}
```

Comment:
Maybe those should be moved to message modules? For example part's definition sightly differs in TC-13-10 and TC-13-11 messages

Reply:
Same for TM messages

Reply:
To be honest I don't see a reason to do that. Could you elaborate?